resources/qml/delegates/ImageMessage.qml

```diff
@@ -68,7 +56,63 @@ AbstractButton {
         eventId: parent.eventId
     }
 
-    onClicked :Settings.openImageExternal ? room.openMedia(eventId) : TimelineManager.openImageOverlay(room, url, eventId, originalWidth, proportionalHeight);
+    Image {
+        id: blurhash_
+
+        anchors.fill: parent
+        visible: img.status != Image.Ready || timeline.privacyScreen.active
+        source: blurhash ? ("image://blurhash/" + blurhash) : ("image://colorimage/:/icons/icons/ui/image-failed.svg?" + Nheko.colors.buttonText)
+        asynchronous: true
+        fillMode: Image.PreserveAspectFit
+        sourceSize.width: parent.width * Screen.devicePixelRatio
+        sourceSize.height: parent.height * Screen.devicePixelRatio
+        state: img.status != Image.Ready ? "Visible" : (timeline.privacyScreen.active ? "Visible" : "Invisible")
+
+        states: [
+            State {
+                name: "Visible"
+
+                PropertyChanges {
+                    target: blurhash_
+                    opacity: 1
+                }
+            },
+            State {
+                name: "Invisible"
+
+                PropertyChanges {
+                    target: blurhash_
+                    opacity: 0
+                }
+            }
+        ]
+        transitions: [
+            Transition {
+                from: "Visible"
+                to: "Invisible"
+
+                NumberAnimation {
+                    target: blurhash_
+                    property: "opacity"
+                    duration: 250
+                    easing.type: Easing.InQuad
```

Comment:
This actually looks a bit jarring. Maybe it is better with an OutQuad or just a longer duration? If you use 500ms, you probably can just switch to an InOutQuad and make the transition reversible to get rid of some code.

Reply:
I made this the same as the privacy screen transition, but I can fiddle around with different transitions.

Reply:
I changed the PrivacyScreen animation (and this one as well) to Easing.Linear, which I think looks fine